object/map.js

```diff
@@ -0,0 +1,9 @@
+if (!Object.prototype.map) {
+  Object.prototype.map = function(callback) {
+    var array = []
+    Object.keys(this).map((key, index) => {
+       array.push(callback(this[key], key, index))
+    });
+    return array
```

Comment:
Why not:

```
return Object.keys(this).map(function(key, index) {
  return callback(this[key], key, index);
});
```

Couple more things:

1. can't use arrow functions here, needs to be all native. 
2. Let's use the `Object.defineProperty` syntax here so it doesn't show up in enumerations.